include/ibamr/IBFEMethod.h

```diff
@@ -767,6 +767,12 @@ class IBFEMethod : public IBStrategy
      */
     bool d_do_log = false;
 
+    /*
+     * Boolean controlling whether or not the scratch hierarchy should be
+     * used.
+     */
+    bool d_use_scratch_hierarchy = false;
```

Comment:
Could this just be determined by whether the scratch hierarchy is null/non-null?

Reply:
Unfortunately no - we cannot set up the scratch hierarchy until partway through the initialization process, so we need something else to indicate whether or not to use it. This boolean is actually set in #654 from a value in the input file.